lib/fieldhand/paginator.rb

```diff
@@ -81,10 +80,14 @@ def request(query = {})
       request_uri.query = encode_query(query)
 
       logger.info('Fieldhand') { "GET #{request_uri}" }
-      http.get(request_uri.request_uri).body
+      res = http.get(request_uri.request_uri)
+      raise NetworkError.new("Invalid response: #{res.code} #{res.msg}", res) unless res.is_a?(::Net::HTTPSuccess)
+
+      res.body
     rescue ::Timeout::Error => e
       raise NetworkError, "timeout requesting #{query}: #{e}"
     rescue => e
+      raise e if e.is_a?(NetworkError)
```

Comment:
I wonder if we could refactor this away by moving the new exception to `parse_response`: change it so that `request` returns the raw `Net::HTTPResponse` rather than the body and then have `parse_response` raise the error if it's not a 200?

This way, we can leave these two rescues here and avoid the re-raising, e.g.

```ruby
def parse_response(query = {})
  response = request(query)
  raise NetworkError, ... unless response.is_a?(::Net::HTTPSuccess)

  response_parser = ResponseParser.new(response)
  response_parser.errors.each do |error|
    raise error
  end
  
  response_parser
end
```

I'm also now wondering whether we should introduce a _new_ error type for this rather than overloading `NetworkError`, maybe `ResponseError` or something like that?